framework/src/problems/FEProblemBase.C

```diff
@@ -1192,10 +1192,20 @@ FEProblemBase::initialSetup()
 
   // Perform Reporter get/declare check
   _reporter_data.check();
+  checkDuplicatePostprocessorVariableNames();
```

Comment:
It looks like @zachmprince moved `_reporter_data.check()` from `checkProblemIntegrity` to here in #17485. My instinct was to say we should move this check to `checkProblemIntegrity` but perhaps this is connected with the check in the line before it. I honestly do not know too much about reporters but I guess they hold postprocessor data these days? Can postprocessors be added during `initialSetup`? Well of course they can be, but I'm just wondering if that is a blessed idiom. It sounds like reporters can be based on #17468, but just looking at the inheritance tree, a `Postprocessor` is not a `Reporter`.

If adding postprocessors during `initialSetup` is not a blessed action, and I would think it shouldn't be, but as I've said I don't understand how postprocessosrs and reporters are related ... then I would like this moved to `checkProblemIntegrity`

Reply:
Okay, good to know -- where does that leave us re: moving the function call though?

Reply:
I read in the postprocessor documentation that people should use `Reporters` instead because they are more general. I guess I am curious why `Postprocessor` doesn't just directly inherit from `Reporter`? Also the reporter documentation says `Reporter` builds off `UserObject` which isn't true in the C++ inheritance sense. But I guess `GeneralReporter` inherits from `GeneralUserObject`.

@zachmprince do you know why `Postprocessor` doesn't inherit from `Reporter`?

Reply:
Semantically it belongs to checkProblemIntegrity() so unless it triggers unreasonable additional work we should try to have it there. 
Alex is more aware of the history and existence of these checks than I am, hence why I did not suggest that earlier